src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs;
+use std::io::Result;
+use std::path::{Path, PathBuf};
+
+pub struct Config {
+    path: PathBuf,
+    root: String,
+}
+
+impl Config {
+    pub fn new(root: impl Into<String>, path: impl Into<String>) -> Self {
```

Comment:
Could you provide some details why the arguments were changed to something which implements the Into trait?

Reply:
Look at my previous comment 😅